### PR TITLE
[opt](Nereids) forbid distribute under project and filter

### PR DESCRIPTION
**Pull request by morrySnow**

### CODE REVIEW

**Comment by morrySnow** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/properties/ChildrenPropertiesRegulator.java [hunk 1]
```diff
@@ -464,6 +467,9 @@ public Boolean visitPhysicalNestedLoopJoin(PhysicalNestedLoopJoin<? extends Plan
     @Override
     public Boolean visitPhysicalProject(PhysicalProject<? extends Plan> project, Void context) {
         // do not process must shuffle
+        if (children.get(0).getPlan() instanceof PhysicalDistribute) {
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/properties/ChildrenPropertiesRegulator.java:470]: if (children.get(0).getPlan() instanceof PhysicalDistribute && project.getOutput().size()  <= children.get(0).getPlan().getOutput().size()) { ...}
how about this?

Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/properties/ChildrenPropertiesRegulator.java:470]: too many check deps on project must under distribute, so we forbid it now until we found a way to remove those check

**Comment by morrySnow** [comment]
run buildall

**Review by englefly — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**